src/serialize.rs

```diff
@@ -1,57 +1,71 @@
-use crate::reduction::EvalErr;
-use std::io::Cursor;
-use std::io::Read;
-use std::io::Seek;
-use std::io::Write;
-use std::io::{Error, ErrorKind, SeekFrom};
+use std::io;
+use std::io::{Cursor, ErrorKind, Read, Seek};
 
 use crate::allocator::{Allocator, NodePtr, SExp};
 use crate::node::Node;
+use crate::reduction::EvalErr;
 
 const MAX_SINGLE_BYTE: u8 = 0x7f;
 const CONS_BOX_MARKER: u8 = 0xff;
 
-fn bad_encoding() -> std::io::Error {
-    Error::new(ErrorKind::InvalidInput, "bad encoding")
+fn bad_encoding() -> io::Error {
+    io::Error::new(ErrorKind::InvalidInput, "bad encoding")
```

Comment:
Usually a bare `Error` refers to the `std::error::Error` trait, so we avoid importing `std::io::Error` here, and use `io::Error` (as per the recommendation for `std::io:Result` found [here](https://doc.rust-lang.org/std/io/type.Result.html).